airflow/jobs/scheduler_job.py

```diff
@@ -721,25 +721,14 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
             if run.state == State.RUNNING:
                 make_transient(run)
                 active_dag_runs.append(run)
-
-        for run in active_dag_runs:
-            self.log.debug("Examining active DAG run: %s", run)
-            tis = run.get_task_instances(state=SCHEDULEABLE_STATES)
-
-            # this loop is quite slow as it uses are_dependencies_met for
-            # every task (in ti.is_runnable). This is also called in
-            # update_state above which has already checked these tasks
-            for ti in tis:
-                task = dag.get_task(ti.task_id)
-
-                # fixme: ti.task is transient but needs to be set
-                ti.task = task
-
-                if ti.are_dependencies_met(
```

Comment:
I'm trhing to think if there's a possible change of behaviour here - the `run.update_state()` was calling with `ignore_in_retry_period=True, ignore_in_reschedule_period=True` but we aren't here.

Reply:
(I do like this change overall though, just need to check it hasn't subtly changed a behaviour.)

Reply:
It did change the behaviour, It could have fail tasks that were up_for_retry and premature. I made another iteration trying to keep the same logic but a bit clearer and cleaner . Please review!